internal/store/store.go

```diff
@@ -35,12 +34,15 @@ func New(reader io.Reader, logger logrus.FieldLogger) (*Store, error) {
 
 func validatePlugins(plugins []*model.Plugin) error {
 	for _, plugin := range plugins {
-		if plugin.Manifest.Id == "" {
-			return errors.Errorf("plugin manifest Id is empty %+v", plugin)
+		err := plugin.Manifest.IsValid()
+		if err != nil {
+			return errors.Wrapf(err, "invalid manifest for plugin %s", plugin.Manifest.Id)
 		}
-		if _, err := semver.Parse(plugin.Manifest.Version); err != nil {
-			return errors.Wrapf(err, "failed to parse manifest version for manifest.Id %s", plugin.Manifest.Id)
+
+		if plugin.Manifest.Version == "" {
```

Comment:
ahhh i think this will break things, we have manifest that does not contain `version`. What we need to check here is if a version is present, is it valid, if there is no version that that is valid too.

Reply:
The previous code did the same: confirm a version is present and error otherwise. There is even a test for this: https://github.com/mattermost/mattermost-marketplace/pull/44/files#diff-b30f396f3719b724710bcd34e4bcb1aaR36

The code is just keeping this behaviour. Are you suggesting to change it?

Reply:
oh whoops, i was thinking `MinServerVersion`. You're right we should always have a version value.

Hmm, is `manifest.version` optional? i thought that field is mandatory? if it is mandatory shouldn't `Manifest.IsValid()` check if the version is there? rather than doing this check here?

Reply:
@ali-farooq0 Well, there is no technical requirement for `manifest.version` right now. If a plugin omits it, the UI just looks a bit weird. 
![Screenshot from 2020-02-05 10-52-23](https://user-images.githubusercontent.com/16541325/73830917-be1f9a80-4805-11ea-8a1f-9682c5bf04e7.png)

Why should it be (technical) mandatory?

Reply:
@hanzei 
that's our only frame of reference to see if a plugin is newer or not, if an author keeps releasing with no version we'll keep overwriting existing plugins in marketplace, no? maybe that could be a requirement for marketplace.